web/src/components/FileListing.js

```
@@ -87,47 +60,55 @@ export default function FileListing({ files, category, searchQuery }) {
-                    <button className="text-indigo-600 hover:text-indigo-900">
-                      Download
-                    </button>
+                    <div className="flex flex-wrap justify-end gap-2">
+                      {config?.actions?.map((action) => (
+                        <button
+                          key={action.label}
+                          className={`rounded-md px-3 py-1 text-xs font-semibold transition duration-150 ${
+                            toneButtonStyles[action.tone] || toneButtonStyles.indigo
+                          }`}
+                        >
+                          {action.label}
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)